### PR TITLE
Fix PRNGKeyArray.broadcast_to with scalar shape

### PR DESCRIPTION
**Pull request by YouJiacheng**

### CODE REVIEW

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good - add a test?

**Comment by YouJiacheng** [comment]
Should I add a test directly use PRNGKeyArray? Or just test with PRNGKey(will not use PRNGKeyArray unless config set custom prng key, while CI doesn't set this.)

**Comment by jakevdp** [comment]
I would probably add the test somewhere near here: https://github.com/google/jax/blob/f8d796939212fd2b3651c10c5e00f2f7503453c2/tests/random_test.py#L1550-L1555

I don't think we test custom_prng on github CI, but we have other test coverage for that that will catch it (and eventually custom_prng will be on by default)

File: tests/random_test.py [hunk 1]
```diff
@@ -1553,6 +1553,9 @@ def test_broadcast_to(self):
     ref = jnp.broadcast_to(like(key), (3,))
     self.assertEqual(out.shape, ref.shape)
     self.assertEqual(out.shape, (3,))
+    out = jnp.broadcast_to(key,        3)
```
Comment (jakevdp) [inline tests/random_test.py:1556]: Use a single space please

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**